src/main/java/io/pravega/connectors/flink/FlinkPravegaTableSource.java

```diff
@@ -78,42 +70,84 @@ public FlinkPravegaTableSource(
      */
     @Override
     public DataStream<Row> getDataStream(StreamExecutionEnvironment env) {
-        FlinkPravegaReader<Row> reader = createFlinkPravegaReader();
+        FlinkPravegaReader<Row> reader = sourceFunctionFactory.get();
+        reader.initialize();
         return env.addSource(reader);
     }
 
+    /**
+     * NOTE: This method is for internal use only for defining a TableSource.
+     *       Do not use it in Table API programs.
+     */
     @Override
-    public TypeInformation<Row> getReturnType() {
-        return typeInfo;
+    public DataSet<Row> getDataSet(ExecutionEnvironment env) {
```

Comment:
If this method is coming from the implemented interfaces, then why is it the case that programs can't use it? I'd say that this needs to be at least package protected, but the compiler won't allow you to do it.

Reply:
I agree that it's a questionable design choice on the part of Flink Table API, but there's nothing I can do.